src/Integration.Vsix/SonarLintTagger/IssueTagger.cs

```diff
@@ -37,21 +37,21 @@ namespace SonarLint.VisualStudio.Integration.Vsix
     /// </remarks>
     internal sealed class IssueTagger : ITagger<IErrorTag>, IDisposable
     {
-        private IssuesSnapshot issues;
+        public delegate void OnTaggerDisposed(IssueTagger issueTagger);
+
+        private IEnumerable<IssueMarker> issueMarkers;
+        private readonly OnTaggerDisposed onTaggerDisposed;
         private bool isDisposed;
-        internal /* for testing */ IIssueTracker IssueTracker { get; }
 
-        public IssueTagger(IIssueTracker issueTracker)
+        public IssueTagger(IEnumerable<IssueMarker> issueMarkers, OnTaggerDisposed onTaggerDisposed)
```

Comment:
Simplification: the tagger no longer needs to know about the tracker. It only needs to know the list of markers to display, and a callback to notify when it is disposed.
I could have used a `Func<>` instead of a delegate, but a delegate is slightly more self-descriptive.